jparest/src/test/java/uk/gov/homeoffice/digital/sas/jparest/controller/ResourceApiControllerTest.java

```diff
@@ -582,6 +583,309 @@ void update_idExistsOnRequestPathAndDoesNotExistOnBody_verifyPayloadIsValidatedW
+        var getResourceTwo = getResponseTwo.getItems().get(0);
+        assertThat(getResourceTwo.getDescription()).isEqualTo("Dummy Entity C Two");
+
+        //create update payload
+        String updatedResourceOnePayload = PayloadCreator.createPayload(Map.of(ID_FIELD_NAME, createdResource.getId(),
+            DESCRIPTION_FIELD_NAME, "Updated Dummy Entity C One",
+            INDEX_FIELD_NAME, 2, TENANT_ID_FIELD_NAME, TENANT_ID));
+
+        String updatedResourceTwoPayload = PayloadCreator.createPayload(Map.of(ID_FIELD_NAME, createdResource2.getId(),
+            DESCRIPTION_FIELD_NAME, "Updated Dummy Entity C Two",
+            INDEX_FIELD_NAME, 2, TENANT_ID_FIELD_NAME, TENANT_ID));
+
+        var updatedPayload = new String[] {updatedResourceOnePayload, updatedResourceTwoPayload};
+
+        var payload = Arrays.toString(updatedPayload);
+
+        var updateResponse = controller.batchUpdate(TENANT_ID, payload);
+
+
+        assertThat(updateResponse.getItems()).hasSize(2);
+
+        var dummyOne = updateResponse.getItems().get(0);
+        assertThat(dummyOne).isNotNull();
+        assertThat(dummyOne.getId()).isEqualTo(createdResource.getId());
+        assertThat(dummyOne.getIndex()).isEqualTo(2);
+        assertThat(dummyOne.getDescription()).isEqualTo("Updated Dummy Entity C One");
+
+        var dummyTwo = updateResponse.getItems().get(1);
+        assertThat(dummyTwo).isNotNull();
+        assertThat(dummyTwo.getId()).isEqualTo(createdResource2.getId());
+        assertThat(dummyTwo.getIndex()).isEqualTo(2);
+        assertThat(dummyTwo.getDescription()).isEqualTo("Updated Dummy Entity C Two");
+
+        var checkResponse = controller.get(TENANT_ID, createdResource.getId());
+        var checkResource = checkResponse.getItems().get(0);
+        assertThat(checkResource).isEqualTo(dummyOne);
+
+        var checkResponse2 = controller.get(TENANT_ID, createdResource2.getId());
+        var checkResource2 = checkResponse2.getItems().get(0);
+        assertThat(checkResource2).isEqualTo(dummyTwo);
+    }
+
+    @ParameterizedTest(name="{0}")
+    @MethodSource("invalidPayloads")
+    @Transactional
+    void batchUpdate_resourceExistsInvalidPayload_jsonExceptionThrown(String payload) {
+        var controller = getResourceApiController(DummyEntityA.class);
+        assertThatExceptionOfType(JsonProcessingException.class).isThrownBy(() -> controller.batchUpdate(TENANT_ID, payload));
+    }
+
+    @ParameterizedTest(name="{0}")
+    @MethodSource("invalidPayloads")
+    @Transactional
+    void batchUpdate_resourceDoesntExistInvalidPayload_jsonExceptionThrown(String payload) {
+
+        var controller = getResourceApiController(DummyEntityA.class);
+
+        assertThatExceptionOfType(ResourceNotFoundException.class).isThrownBy(() -> controller.get(TENANT_ID, NON_EXISTENT_ID));
+        assertThatExceptionOfType(JsonProcessingException.class).isThrownBy(() -> controller.batchUpdate(TENANT_ID, payload));
```

Comment:
Based on the equivalent update (single) test I took the test to mean "even if the resource id doesn't exist we'll still get a `JsonProcessingException` if the payload is dodgy".
I'm not entirely sure if the exact same test is relevant here as we don't pass in id/s as a separate request param. 
It looks to me as if like 669 has no relevance to line 670 as the `NON_EXISTENT_ID` isn't part of the payload.
Or maybe I'm not understanding the test correctly..

Reply:
No, my bad, I'd misunderstood the test. I've removed it and amended the name of the one above to reflect the change